packages/scripts/src/class-release.php

```diff
@@ -0,0 +1,261 @@
+		}
+		return true;
+	}
+
+	/**
+	 * Handles root package update
+	 *
+	 * @param String $package package name.
+	 */
+	public function handle_package_update( $package ) {
+		$name = $package['name'];
+		$sh   = new Git_Shell_Command( $name );
+
+		$sh->clone_repository();
+
+		$tag  = $sh->get_latest_tag();
+		$diff = $sh->get_diff_between( $tag, 'master' );
+
+		if ( $this->is_update_requested( $name, $tag, $diff ) ) {
+			$this->do_package_update( $package );
+		}
+	}
+
+	/**
+	 * Loops through provided list of dependencies and tries to update them
+	 *
+	 * @param Array $list list of dependencies.
+	 */
+	public function handle_dependencies_updates( $list ) {
+		Logger::info( 'Here is the list of dependencies with some unreleased changes:' );
+		Logger::log( $list, 1 );
+
+		foreach ( $list as $dep ) {
+			if ( $this->is_update_requested( $dep[0], $dep[2], $dep[1] ) ) {
+				$this->do_package_update( $dep[3] );
+			}
+		}
+	}
+
+	/**
+	 * Run an actual package update
+	 *
+	 * @param String $package package name.
+	 * @throws \Exception Invalid provided version.
+	 */
+	public function do_package_update( $package ) {
+		$name = $package['name'];
+		Logger::info( "Updating package: $name" );
+
+		$prompt = $this->logger->get_colored_string(
+			'Please provide a version number that should be used for new release in SemVer format (x.x.x): ',
+			'green'
+		);
+
+		$version = readline( $prompt );
+
+		if ( 3 !== count( explode( '.', $version ) ) ) {
+			throw new \Exception( "$version is not in a SemVer format" );
+		}
+
+		$version = 'v' . $version;
+		$branch  = "release-$version";
+
+		$sh = new Git_Shell_Command( $name );
+		$sh->clone_repository();
+		$sh->checkout_new_branch( $branch );
+
+		$is_updated = $this->handle_composer_dependencies( $package );
+
+		// If composer deps were updated, we need to commit changed composer.json.
+		if ( $is_updated ) {
+			$git_status = $sh->status();
+			if ( 0 !== strlen( $git_status ) ) {
+				Logger::info( 'Git directory is not clean' );
+				Logger::info( $git_status );
+
+				$answer = $this->handle_polar_question( 'Do you want to commit these changes [y/n]? ' );
+
+				if ( true === $answer ) {
+					$sh->commit();
+				}
+			}
+		}
+
+		$sh->tag_new_version( $version );
+
+		$answer = $this->handle_polar_question( 'All set! We are ready to push a new release. Proceed [y/n]? ' );
+		if ( true === $answer ) {
+			$sh->push_to_remote( $branch );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Checks if user requested an update
+	 *
+	 * @param String $name repo name.
+	 * @param String $tag latest released version.
+	 * @param String $diff short diff of unreleased changes.
+	 */
+	public function is_update_requested( $name, $tag, $diff ) {
+		Logger::log( "Package name: $name", 'blue' );
+		Logger::log( "Latest stable version: $tag", 'blue' );
+		Logger::log( "Unreleased changes: $diff", 'blue' );
+
+		return $this->handle_polar_question( 'Do you want to update this package [y/n]? ' );
```

Comment:
For some reason this one didn't properly output the colored string, maybe because of the `[y/N]` prompt and the symbols inside it?
Here's what I got:
```
[0;32mDo you want to update this package [y/n]? [0m
```

Reply:
It was working fine in my tests (zsh, macOS). Maybe it's a Linux thing? If yes, can I ask you to check if round braces `()` will fix it?

Reply:
No, it's the same, but the colors are escaped properly this time - at least the color returns to normal :)